app.go

```diff
@@ -30,13 +30,21 @@ type App struct {
 	// Enable appsecret proof in every API call to facebook.
 	// Facebook document: https://developers.facebook.com/docs/graph-api/securing-requests
 	EnableAppsecretProof bool
+
+	// The session to send request when parsing tokens or code.
+	// If it's not set, default session will be used.
+	session *Session
 }
 
 // New creates a new App and sets app id and secret.
-func New(appID, appSecret string) *App {
+func New(appID, appSecret string, session *Session) *App {
```

Comment:
We should not make breaking change in public API. I recommend to mark the session as public, that is, `Session *Session` in `App`.

We can add a new private method `session() *Session` in `App` to return a session instance for all methods using session.

Reply:
The problem with making it public is that you will no longer have access to the function `sendOauthRequest()` as its private. Can I suggest Maybe Add a function then `SetSession(*Session)` that will handle this?